toolchain/parse/node_kind.h

```diff
@@ -119,20 +145,69 @@ class NodeKind::Definition : public NodeKind {
   Definition(const Definition&) = delete;
   auto operator=(const Definition&) -> Definition& = delete;
 
+  // Returns true if the node is bracketed; otherwise, child_count is used.
+  constexpr auto has_bracket() const -> bool { return bracket_ != *this; }
+
+  // Returns the bracketing node kind for the current node kind. Requires that
+  // has_bracket is true.
```

Comment:
Note that with your changes to allow both, "requires" comments are now incorrect.

Reply:
Fixed the comment on `child_count` (this one is still correct).